code/game/machinery/ARES/ARES_interface_admin.dm

```diff
@@ -231,6 +231,23 @@
 		logged_access += list(current_ticket)
 	data["access_tickets"] = logged_access
 
+	var/list/security_vents = list()
+	for(var/obj/structure/pipes/vents/pump/no_boom/gas/vent in linked_vents)
+		if(!vent.vent_tag)
+			vent.vent_tag = "Security Vent #[tag_num]"
+			tag_num++
+
+		var/list/current_vent = list()
+		var/is_available = TRUE
+		if(!COOLDOWN_FINISHED(vent, vent_trigger_cooldown))
+			is_available = FALSE
+		current_vent["vent_tag"] = vent.vent_tag
+		current_vent["ref"] = "\ref[vent]"
+		current_vent["available"] = is_available
+		security_vents += list(current_vent)
+
+	data["security_vents"] = security_vents
```

Comment:
You're doing this same logic in 4 different places. Wouldn't it be better to just make a proc in this file that is like `datum/ares_link/proc/get_vent_ui_data()` or something that returns this list you keep creating?